src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectionTestWithSSLCert/CertificateTest.cs

```diff
@@ -0,0 +1,259 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+using System.ServiceProcess;
+using System.Text;
+using Microsoft.SqlServer.TDS.Servers;
+using Microsoft.Win32;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests
+{
+    public class CertificateTest : IDisposable
+    {
+        #region Private Fields
+        private const string IPV4 = @"127.0.0.1";
+        private const string IPV6 = @"::1";
+        private static readonly string s_fullPathToPowershellScript = Path.Combine(Directory.GetCurrentDirectory(), "SQL", "ConnectionTestWithSSLCert", "GenerateSelfSignedCertificate.ps1");
+        private const string LocalHost = "localhost";
+        private static readonly string s_fQDN = Dns.GetHostEntry(Environment.MachineName).HostName;
+        private readonly string _thumbprint;
+        private const string ThumbPrintEnvName = "Thumbprint";
+
+        public string ForceEncryptionRegistryPath
+        {
+            get
+            {
+                if (DataTestUtility.IsSQL2022())
+                {
+                    return @"SOFTWARE\Microsoft\Microsoft SQL Server\MSSQL16.MSSQLSERVER\MSSQLServer\SuperSocketNetLib";
+                }
+                if (DataTestUtility.IsSQL2019())
+                {
+                    return @"SOFTWARE\Microsoft\Microsoft SQL Server\MSSQL15.MSSQLSERVER\MSSQLServer\SuperSocketNetLib";
+                }
+                if (DataTestUtility.IsSQL2016())
+                {
+                    return @"SOFTWARE\Microsoft\Microsoft SQL Server\MSSQL14.MSSQLSERVER\MSSQLServer\SuperSocketNetLib";
+                }
+                return string.Empty;
+            }
+        }
+        #endregion
+
+        public CertificateTest()
+        {
+            CreateValidCertificate(s_fullPathToPowershellScript);
+            _thumbprint = Environment.GetEnvironmentVariable(ThumbPrintEnvName, EnvironmentVariableTarget.Machine);
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup), nameof(DataTestUtility.IsAdmin), nameof(DataTestUtility.IsNotAzureServer))]
```

Comment:
Checking the admin permission would silently skipped the test. I'd check it inside the test and throw exception instead.
Same note for all the consumptions.

Reply:
we want to skip the test if it is not running as admin. not sure if throwing exception would help here.